### PR TITLE
Further copyright updates

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
Somewhat error-prone, so please review carefully.

Reasons we need this:

- Some file types are not properly handled by the script.
- The only exclusion mechanism we currently have (`NO_AUTO_COPYRIGHT`)
  is overly coarse.

CHANGELOG_BEGIN
CHANGELOG_END

### CODE REVIEW

File: compatibility/BUILD [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+# Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
```
Comment (cocreature) [inline compatibility/BUILD:1]: why do we not hit this via automation? I don’t see a no autocopyright file here.

File: compiler/damlc/tests/daml-test-files/InterfaceNotSupported.daml [hunk 2]
```diff
@@ -1,4 +1,4 @@
--- Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
```
Comment (cocreature) [inline compiler/damlc/tests/daml-test-files/InterfaceNotSupported.daml:1]: we should either drop the no autocopyright from this directory (not sure why it’s there in the first place) or drop all the copyright headers. Not in this PR though.

File: daml-lf/encoder/src/test/lf/Decimal_all_.lf [hunk 3]
```diff
@@ -1,4 +1,4 @@
-// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
```
Comment (cocreature) [inline daml-lf/encoder/src/test/lf/Decimal_all_.lf:1]: why is automation missing these files?

**Review by cocreature — APPROVED** [review]
Thanks!

File: compatibility/BUILD (hunk 1 above)
Comment (garyverhaegen-da) [reply compatibility/BUILD:1]: We detect file types (in order to figure out what kind of comment marker to use) based on extension, and this filename does not have an extension.

File: daml-lf/encoder/src/test/lf/Decimal_all_.lf (hunk 3 above)
Comment (garyverhaegen-da) [reply daml-lf/encoder/src/test/lf/Decimal_all_.lf:1]: `lf` is not in the list of supported extensions.

Comment (cocreature) [reply daml-lf/encoder/src/test/lf/Decimal_all_.lf:1]: Can we add it? Or is that some thirdparty dep that we cannot easily change?

File: compiler/damlc/tests/daml-test-files/InterfaceNotSupported.daml (hunk 2 above)
Comment (garyverhaegen-da) [reply compiler/damlc/tests/daml-test-files/InterfaceNotSupported.daml:1]: FWIW removing the `NO_AUTO_COPYRIGHT` file results in the copyright header script crashing on an invalid unicode file. I believe it is intentional that the file is invalid (to test how the compiler reacts on that), and I guess it was easier to ignore the entire folder than fix the script (hence my "too coarse-grained" comment).

File: daml-lf/encoder/src/test/lf/Decimal_all_.lf (hunk 3 above)
Comment (garyverhaegen-da) [reply daml-lf/encoder/src/test/lf/Decimal_all_.lf:1]: I'll see what I can do. I've been reluctant to touch a script that only really annoys me once a year, but this should be simple enough. Separate PR, though.

Comment (garyverhaegen-da) [reply daml-lf/encoder/src/test/lf/Decimal_all_.lf:1]: #12259

**Pull request merged**